pkg/operator/ceph/cluster/controller.go

```diff
@@ -538,13 +587,30 @@ func (c *ClusterController) onUpdate(oldObj, newObj interface{}) {
 	cluster.Spec = &newClust.Spec
 
 	// If the image version changed let's make sure we can safely upgrade
-	// check ceph's status, if not healthy we fail
 	if versionChanged {
-		cephStatus := client.IsCephHealthy(c.context, cluster.Namespace)
-		if !cephStatus {
-			logger.Errorf("ceph status in namespace %s is not healthy, refusing to upgrade. fix the cluster and re-edit the cluster CR to trigger a new orchestation update", cluster.Namespace)
+		// Get cluster running versions
+		versions, err := client.GetCephVersions(c.context, cluster.Namespace)
```

Comment:
Note that in #3450 I am refactoring the common code in onAdd and onUpdate so we can have consistent version checks without the duplicate code. I'll add this as well after it merges.